pseXperiments/freedit
# chore: scaffold monorepo

Towards replacing the template rendering engine used for the web interface by a JS/TS application.  
In the internal PSE #dogfood, it seems we were all in favor on using a monorepo to manage the freedit rs backend and the to be developed JS client.  

I picked a few tools that I am setting up in this PR:
- [`bun`](https://bun.sh/docs/installation) as JS runtime and node package manager
- [`just`](https://just.systems) as tasks runner
- [`vite`](https://vite.dev/): JS/TS front end build tool
- [tanstack router](https://tanstack.com/router/latest/docs/framework/react/overview): client side router
  Right now I am just "dangerously setting" the inner html in one route of client app from what the server gives. But of course we should not keep doing that way, this is just for demo/test purposes atm. Indeed atm the server only serves html documents (the rinja template rendering engine). Eventually we want to have a server that gives us JSON and fetch that client side, and let react build the html docs (see #5 and #6).
  
  ![image](https://github.com/user-attachments/assets/2bbe25d7-7509-4169-a078-3e131ab560b9)


## Test plan
- Spin up the server and client: eg with `just dev`
- you should not see any errors or warnings
- go to [localhost:5173](http://localhost:5173) . Click on the inn link. You should see the same content  as on [localhost:3001](http://localhost:3001) (but displayed very ugly)
